src/ui/components/DeckBuilder/DeckBuilder.test.tsx

```diff
@@ -0,0 +1,112 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { vi } from 'vitest'
+
+import { ICard } from '../../../game/types'
+
+import { DeckBuilder } from './DeckBuilder'
+
+// Mock configuration
+vi.mock('../../../game/config', () => ({
+  DECK_SIZE: 2,
+}))
+
+const { mockCarrot, mockPumpkin, mockWater, mockShovel, mockRain } = vi.hoisted(
+  () => ({
+    mockCarrot: {
+      id: 'carrot',
+      name: 'Carrot',
+      type: 'CROP',
+      waterToMature: 2,
+    },
+    mockPumpkin: {
+      id: 'pumpkin',
+      name: 'Pumpkin',
+      type: 'CROP',
+      waterToMature: 1,
+    },
+    mockWater: { id: 'water', name: 'Water', type: 'WATER' },
+    mockShovel: { id: 'shovel', name: 'Shovel', type: 'TOOL' },
+    mockRain: { id: 'rain', name: 'Rain', type: 'EVENT' },
+  })
+)
```

Comment:
Do we need to define these bespoke mock cards, or can we reuse the stubs from `src/test-utils/stubs/cards.ts` instead?

Either way, update `AGENTS.md` to favor reusing those stubs in tests whenever possible.